calico_mesos/tests/unit/mesos_test.py

```diff
@@ -330,23 +327,46 @@ def test_distpach_calls_release(self, m_stdin, m_release):
 
 
 class TestDefaultProfile(unittest.TestCase):
+    HOST_IP_NET = "172.16.0.0/16"
+
+    @patch('calico_mesos._get_host_ip_net', side_effect=lambda: TestDefaultProfile.HOST_IP_NET)
     @patch('calico_mesos.datastore')
-    def test_default_profile(self, m_datastore):
-        m_update_rules = Mock()
-        m_datastore.profile_update_rules = m_update_rules
+    def test_correct_rules_for_host_profile(self, m_datastore, m_get_host_ip_net):
+        calico_mesos._create_profile_for_host_communication("default")
+        new_rules = m_datastore.get_profile().rules
+        self.assertIn(Rule(action="allow", src_net=self.HOST_IP_NET), new_rules.inbound_rules)
+        self.assertIn(Rule(action="allow", dst_net=self.HOST_IP_NET), new_rules.outbound_rules)
+        self.assertEqual(len(new_rules.inbound_rules) + len(new_rules.outbound_rules), 2)
 
-        m_profile = Mock()
-        m_datastore.get_profile.return_value = m_profile
+    @patch('calico_mesos.datastore')
+    def test_correct_rules_for_netgroup_profile(self, m_datastore):
+        calico_mesos._create_profile_for_netgroup("prof_a")
+        new_rules = m_datastore.get_profile().rules
+        self.assertIn(Rule(action="allow", src_tag="prof_a"), new_rules.inbound_rules)
+        self.assertIn(Rule(action="allow"), new_rules.outbound_rules)
+        self.assertEqual(len(new_rules.inbound_rules) + len(new_rules.outbound_rules), 2)
 
-        calico_mesos._create_profile_with_default_mesos_rules("TESTPROF")
+    @patch('calico_mesos.datastore')
+    def test_correct_rules_for_public_profile(self, m_datastore):
+        calico_mesos._create_profile_for_public_communication("public")
+        new_rules = m_datastore.get_profile().rules
+        self.assertIn(Rule(action="allow"), new_rules.inbound_rules)
+        self.assertIn(Rule(action="allow"), new_rules.outbound_rules)
+        self.assertEqual(len(new_rules.inbound_rules) + len(new_rules.outbound_rules), 2)
 
-        new_rules = m_profile.rules
 
-        host_net = calico_mesos._get_host_ip_net()
-        #TODO: Better test for getting host ip
-        self.assertIn(Rule(action="allow", src_net=host_net), new_rules.inbound_rules)
-        self.assertIn(Rule(action="allow", src_tag="TESTPROF"), new_rules.inbound_rules)
-        self.assertIn(Rule(action="allow"), new_rules.outbound_rules)
+    @patch('calico_mesos.datastore')
+    def test_profiles_are_created(self, m_datastore):
+        m_datastore.profile_exists.side_effect = lambda x: False
+
+        profiles = ["public", "prof_a"]
+        calico_mesos._isolate("testhostname", 1234, "container-id-1234", ["192.168.0.0"], [], profiles, None)
+        created_endpoint = m_datastore.create_endpoint()
```

Comment:
it'd be a better test to do 
```
created_endpoint = Mock(spec=Endpoint)
m_datastore.create_endpoint.return_value = m_endpoint
```

before the call to `_isolate`.  That way you can catch bugs where `_isolate` (or this test!) mistypes the API of `Endpoint`.